loadtest/store/memstore/store.go

```diff
@@ -64,6 +64,33 @@ func (s *MemStore) SetChannel(channel *model.Channel) error {
 	return nil
 }
 
+func (s *MemStore) Team(teamId string) (*model.Team, error) {
+	if team, ok := s.teams[teamId]; ok {
+		return team, nil
+	}
+	return nil, nil
+}
+
+func (s *MemStore) SetTeam(team *model.Team) error {
+	s.teams[team.Id] = team
+	return nil
+}
+
+func (s *MemStore) Teams() ([]*model.Team, error) {
+	teams := []*model.Team{}
+	for _, team := range s.teams {
+		teams = append(teams, team)
+	}
+	return teams, nil
+}
+
+func (s *MemStore) SetTeams(teams []*model.Team) error {
+	for _, team := range teams {
+		s.teams[team.Id] = team
+	}
+	return nil
+}
```

Comment:
This can cause the state to get *polluted*. With the above function we always either add or update and never delete. I think the safest choice right now is to make all store setters always replace the data. In this case we could either replace the map entirely or empty it before setting the teams.

Reply:
I'm ok on with the approach, but then we would have to either create update methods or allow updating on some to avoid code repetition. E.g.: after a `SearchUsers` action we cannot use `SetUsers` with the results as it would remove the existing ones that were saved previously on (for example) the `GetUsersForChannel` action.

We can solve this with having both `SetUsers` and `AddUsers`/`UpdateUsers`, for example, to allow for both paths, iterate through results and use `SetUser` (suboptimal) or have a flag that indicates if we want the method to replace preexisting users or not. Any preference here?

Reply:
Sure. I am fine with additional `Add`/`Remove` methods if it comes to that, but let's try to not write code we don't yet need. For example, the case of searching the users could be one of those cases where we only act on the API result and don't store anything.